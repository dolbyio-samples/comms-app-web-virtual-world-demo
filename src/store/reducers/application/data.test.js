import { expect, it, describe } from "vitest";
import reducer from "./data";

describe("data reducer test suite", () => {
  it("should update the state - curren session id", () => {
    // arrange
    const action = {
      type: "OPENED",
      payload: {
        id: "sessionID",
      },
    };

    const state = {};

    // act
    const updatedState = reducer(state, action);

    // assert
    expect(updatedState).toMatchSnapshot();
  });

  it("should update the state - current participant id", () => {
    // arrange
    const action = {
      type: "SET_LOCAL_PARTICIPANT_ID",
      payload: {
        id: "participantId",
      },
    };

    const state = {};

    // act
    const updatedState = reducer(state, action);

    // assert
    expect(updatedState).toMatchSnapshot();
  });

  it("should update the state - current conference id", () => {
    // arrange
    const action = {
      type: "SET_CURRENT_CONFERENCE_ID",
      payload: {
        id: "conferenceId",
      },
    };

    const state = {};

    // act
    const updatedState = reducer(state, action);

    // assert
    expect(updatedState).toMatchSnapshot();
  });

  it("should update the state - current device id", () => {
    // arrange
    const action = {
      type: "SET_CURRENT_DEVICE_ID",
      payload: {
        id: "deviceId",
      },
    };

    const state = {};

    // act
    const updatedState = reducer(state, action);

    // assert
    expect(updatedState).toMatchSnapshot();
  });

  it("should update the state - message id", () => {
    // arrange
    const action = {
      type: "ADD_MESSAGE_ID",
      payload: {
        id: "messageId",
      },
    };

    const state = {};

    // act
    const updatedState = reducer(state, action);

    // assert
    expect(updatedState).toMatchSnapshot();
  });

  it("should update the state - remove message id", () => {
    // arrange
    const action = {
      type: "REMOVE_MESSAGE_ID",
      payload: {
        id: "4",
      },
    };

    const state = { messages: ["1", "2", "3", "4"] };

    // act
    const updatedState = reducer(state, action);

    // assert
    expect(updatedState).toMatchSnapshot();
  });

  it("should update the state - clean state", () => {
    // arrange
    const action = {
      type: "LEAVE_CONFERENCE",
      payload: {
        id: "conferenceId",
      },
    };

    const state = {
      currentDeviceID: "123",
      messages: ["1"],
      currentConferenceID: "768",
      currentSessionID: "785",
    };

    // act
    const updatedState = reducer(state, action);

    // assert
    expect(updatedState).toMatchSnapshot();
  });
});
