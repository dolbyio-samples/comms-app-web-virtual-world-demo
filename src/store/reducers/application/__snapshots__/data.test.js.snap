// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`data reducer test suite > should update the state - clean state 1`] = `
{
  "currentDeviceID": "123",
  "currentSessionID": "785",
}
`;

exports[`data reducer test suite > should update the state - curren session id 1`] = `
{
  "currentSessionID": "sessionID",
}
`;

exports[`data reducer test suite > should update the state - current conference id 1`] = `
{
  "currentConferenceID": "conferenceId",
}
`;

exports[`data reducer test suite > should update the state - current device id 1`] = `
{
  "currentDeviceID": "deviceId",
}
`;

exports[`data reducer test suite > should update the state - current participant id 1`] = `
{
  "localParticipantID": "participantId",
}
`;

exports[`data reducer test suite > should update the state - message id 1`] = `
{
  "messages": [
    "messageId",
  ],
}
`;

exports[`data reducer test suite > should update the state - remove message id 1`] = `
{
  "messages": [
    "1",
    "2",
    "3",
  ],
}
`;
